site/content/guides/grpc.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Configuring ingress to gRPC services with Contour
+layout: page
+---
+
+## Example gRPC Service
+
+The below examples use the [gRPC server](https://github.com/projectcontour/yages) used in Contour end to end tests.
+The server implements a service `yages.Echo` with two methods `Ping` and `Reverse`.
+It also implements the [gRPC health checking service](https://pkg.go.dev/google.golang.org/grpc/health/grpc_health_v1) (see [here](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) for more details) and is bundled with the [gRPC health probe](https://github.com/grpc-ecosystem/grpc-health-probe).
+
+An example base deployment and service for a gRPC server utilizing plaintext HTTP/2 are provided here:
+
+```yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/name: grpc-echo
+  name: grpc-echo
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: grpc-echo
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: grpc-echo
+    spec:
+      containers:
+      - name: grpc-echo
+        image: ghcr.io/projectcontour/yages:v0.1.0
+        ports:
+        - name: grpc
+          containerPort: 9000
+        readinessProbe:
+          exec:
+            command: ["/grpc-health-probe", "-addr=localhost:9000"]
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app.kubernetes.io/name: grpc-echo
+  name: grpc-echo
+spec:
+  selector:
+    app.kubernetes.io/name: grpc-echo
+  ports:
+  - port: 9000
+    protocol: TCP
+    targetPort: grpc
+```
+
+## HTTPProxy Configuration
+
+Configuring proxying to a gRPC service with HTTPProxy is as simple as specifying the protocol Envoy uses with the upstream application via the `spec.routes[].services[].protocol` field.
```

Comment:
Do we also want to suggest setting the route response timeout to a large value (ref. https://github.com/projectcontour/contour/issues/3692#issuecomment-1004192741)?

Reply:
yee definitely, good guideline to add for streaming RPCs